src/utilities/valueFromAST.js

```diff
@@ -152,9 +152,11 @@ export function valueFromAST(
   );
 
   const parsed = type.parseLiteral(valueNode);
-  if (isNullish(parsed)) {
-    // null or invalid values represent a failure to parse correctly,
-    // in which case no value is returned.
+  if (type instanceof GraphQLEnumType ?
+      !type.serialize(parsed) : isNullish(parsed)) {
```

Comment:
Hmm this condition is a little gross now. I wonder if we should instead have something like `type.isValidValue()` or something along those lines to simplify this?